src/include/flow.h

```diff
@@ -646,8 +646,12 @@ class Flow {
     machneth->msg_flags = msg_flags;
     machneth->seqno = be32_t(seqno);
     machneth->ackno = be32_t(pcb_.ackno());
-    machneth->sack_bitmap = be64_t(pcb_.sack_bitmap);
+
+    for (uint8_t i = 0; i < pcb_.kSackBitmapSize / 64; ++i) {
```

Comment:
Hardcoded `64` should be replaced.

In particular since you assigning to `macneth->sack_bitmap` you should enumerate based on this size.